core/src/com/projectkorra/projectkorra/waterbending/OctopusForm.java

```diff
@@ -176,10 +176,10 @@ private void form() {
 		if (isPlant(this.sourceBlock) || isSnow(this.sourceBlock)) {
 			new PlantRegrowth(this.player, this.sourceBlock);
 			this.sourceBlock.setType(Material.AIR);
-		} else if (!GeneralMethods.isAdjacentToThreeOrMoreSources(this.sourceBlock) && this.sourceBlock != null && !isCauldron(this.sourceBlock)) {
+		} else if (!GeneralMethods.isAdjacentToThreeOrMoreSources(this.sourceBlock) && this.sourceBlock != null && !isNonTransparentSource(this.sourceBlock)) {
 			this.sourceBlock.setType(Material.AIR);
-		} else if (isCauldron(this.sourceBlock)) {
-			GeneralMethods.setCauldronData(this.sourceBlock, ((Levelled) this.sourceBlock.getBlockData()).getLevel() - 1);
+		} else if (isCauldron(this.sourceBlock) || isMud(this.sourceBlock) || isSponge(this.sourceBlock)) {
+			updateSourceBlock(this.sourceBlock);
 		}
```

Comment:
This should use the map too right?

Reply:
There's an issue of whether or not we want users to be able to freely change the keys of transformable blocks in the config, hence we fix our conditions to only mud, sponges and cauldrons. If we do want to allow users complete freedom over transformable blocks, yes, we should check with our map.

Reply:
If we dont check the map atm someone could remove the mud and or sponge transformations and it would still play sounds but nothing would change, albiet that should probably be addressed in the method itself

Reply:
Regarding this, yeah. Most changes can be made in the `updateSourceBlock` function. Though I worry I may need to update the conditions in each ability again when we support other stuff like mossy cobblestone. I think in that case I'll just check with some `isTransformableBlock` function in `WaterAbility` for every ability now rather than just keep on piling or statements.

Reply:
- Addressed in commit https://github.com/ProjectKorra/ProjectKorra/pull/1329/commits/5c4dd4c8c9e00a4c246d5f9009d3878ebea5e6bd.